core/src/main/scala/org/apache/spark/rpc/akka/AkkaRpcEnv.scala

```diff
@@ -214,8 +213,10 @@ private[spark] class AkkaRpcEnv private[akka] (
 
   override def asyncSetupEndpointRefByURI(uri: String): Future[RpcEndpointRef] = {
     import actorSystem.dispatcher
-    actorSystem.actorSelection(uri).resolveOne(defaultLookupTimeout).
-      map(new AkkaRpcEndpointRef(defaultAddress, _, conf))
+    defaultLookupTimeout.addMessageIfTimeout(
+      actorSystem.actorSelection(uri).resolveOne(_).
+        map(new AkkaRpcEndpointRef(defaultAddress, _, conf))
+    )
```

Comment:
yeah I think its better to be more explicit, its not clear what the first `_` is.  Also I think it would be a bit clearer in two statements, so you can add a comment explaining, eg.

```
val future = actorSystem.actorSelection(uri).resolveOne(defaultLookupTimeout.duration)
  .map(new AkkaRpcEndpointRef(defaultAddress, _, conf))
// this is just in case there is a timeout from creating the future in resolveOne, we want the
// exception to indicate the conf that determines the timeout
defaultLookupTimeout.addMessageIfTimeout(future)
```